dask_ml/model_selection/_incremental.py

```diff
@@ -522,7 +522,10 @@ def _validate_parameters(self, X, y):
         kwargs = dict(accept_unknown_chunks=False, accept_dask_dataframe=False)
         X = self._check_array(X, **kwargs)
         y = self._check_array(y, ensure_2d=False, **kwargs)
-        scorer = check_scoring(self.estimator, scoring=self.scoring)
+        estimator = self.estimator
+        if isinstance(estimator, Future):
+            estimator = await estimator.result()
```

Comment:
PyTorch and Keras models are often large.* When this model is put into the DAG, a warning is printed saying "Use data_future = client.scatter(data) and pass that to client.submit." This code allows `client.scatter` to take a model future instead of a (large) model.

I don't know if it's necessary to call `check_scoring`. In the case of PyTorch/Keras models it's probably not necessary. If it is, it won't be so bad because `check_scoring` only checks for attributes of `estimator`. It might not be necessary to bring the entire model to the client; I think we could use the implementation below:

<details>
<summary>Check for attrs only</summary>

``` python
from types import SimpleNamespace
def getattrs(cls):
    return [x for x in dir(cls) if x[0] != "_" and callable(getattr(cls, x))]

def _validate_params(self, X, y):
    ...
    attrs = await client.compute(getattrs, self)
    kwargs = {a: lambda x: for a in attrs}
    dummy = SimpleNamespace(**kwargs)
    scorer = check_scoring(estimator, scoring=self.scoring)
    return X, y, scorer
```

</details>

On model size: very small neural nets have at least 100,000 parameters, or 360KB. Small neural nets have about 1 million parameters, about 3.8MB. The smallest deep network has 11 million parameters, or 44MB:

``` python
>>> import torchvision.models as m
>>> sum([x.nelement() for x in m.resnet18.parameters()]) / 1e6
11.689512
```

Reply:
Would it be possible to keep the estimator remote in this case and send the `check_scoring` function up?

```python
scorer = await client.submit(check_scoring, estimator, scoring=self.scoring)
```

I've been playing with cloud systems from my laptop a lot recently (coiled stuff) and I've started to become very sensitive to sending things back to the client.  My use case may be uncommon today, so we don't need to pay too much attention here, but it seems like it might be easy and slightly more efficient.

Reply:
D'oh. Thanks for the suggestion @mrocklin!